server/models/LessonModel.mjs

```diff
@@ -42,16 +38,16 @@ class LessonModel {
   }
 
   // Convert from Firestore format
-  static fromFirestore(data) {
-    return new LessonModel({
-      lessonId: data.lessonId,
+  static fromFirestore(data, lessonId) {
+    return {
+      lessonId: lessonId,  // Firestore document ID
       title: data.title,
       description: data.description || {}, // Ensure default to empty object if not present
       points: data.points || 0, // Ensure default to 0 if not present
       order: data.order || 0, // Ensure default to 0 if not present
       videoLinks: data.videoLinks || [], // Ensure default to empty array if not present
       materials: data.materials || '' // Ensure default to empty string if not present
-    });
+    };
   }
 }
 
```

Comment:
Good changes Edith, you did important updates to the courseController, lessonController. Thank you!